CHANGELOG.md

```diff
@@ -34,6 +34,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Add query for initialized extensions ([#5658](https://github.com/opensearch-project/OpenSearch/pull/5658))
 - Revert 'Added jackson dependency to server' and change extension reading ([#5768](https://github.com/opensearch-project/OpenSearch/pull/5768))
 - Add support to disallow search request with preference parameter with strict weighted shard routing([#5874](https://github.com/opensearch-project/OpenSearch/pull/5874))
+- Replace latches with CompletableFutures for extensions ([#5646](https://github.com/opensearch-project/OpenSearch/pull/5646))
```

Comment:
@ryanbogan Can you move this entry to the `[Unreleased 2.x]` section of this file since this is going to be released in the next 2.x version? The fact that this entry was in the wrong spot is possibly the reason why the cherry pick had conflicts on the backport. See the [contributing doc](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#where-should-i-put-my-changelog-entry) for details about where to place your changelog entry.